.azure-pipelines/test_plan.py

```diff
@@ -276,6 +277,9 @@ def create(self, topology, test_plan_name="my_test_plan", deploy_mg_extra_params
 
         # If triggered by the internal repos, use internal sonic-mgmt repo as the code base
         sonic_mgmt_repo_url = GITHUB_SONIC_MGMT_REPO
+        if "msft" in repo_name:
+            sonic_mgmt_repo_url = GITHUB_SONIC_MGMT_REPO_MSFT
+
         if kwargs.get("source_repo") in INTERNAL_REPO_LIST:
             sonic_mgmt_repo_url = INTERNAL_SONIC_MGMT_REPO
```

Comment:
It's better to use constants like below if, and because these two if conditions are mutually exclusive(both update sonic_mgmt_repo_url), it is recommended to use if/elif

Reply:
And I found `repo_name` is not defined in this method, looks like use from outside, suggest to use `kwargs.get("source_repo")` as well.

Reply:
modified